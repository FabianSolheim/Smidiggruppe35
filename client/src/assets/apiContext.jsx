import React from "react";
import { fetchJSON } from "./fetchJSON";
import { postJSON } from "./postJSON";

export const ApiContext = React.createContext({
  async listFaq(query) {
    return await fetchJSON("/api/faqs?" + new URLSearchParams(query));
  },
});
