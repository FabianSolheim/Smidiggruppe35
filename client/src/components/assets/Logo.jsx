import React from 'react'

const Logo = () => {
  return (
    <>
    <svg width="122" height="121" viewBox="0 0 122 121" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M48.4438 21.6219C53.7493 21.6219 58.0502 17.247 58.0502 11.8503C58.0502 6.45353 53.7493 2.07861 48.4438 2.07861C43.1383 2.07861 38.8374 6.45353 38.8374 11.8503C38.8374 17.247 43.1383 21.6219 48.4438 21.6219Z" fill="#EED655"/>
<path d="M28.8203 23.7013C32.997 23.7013 36.3828 20.2572 36.3828 16.0087C36.3828 11.7602 32.997 8.31616 28.8203 8.31616C24.6437 8.31616 21.2578 11.7602 21.2578 16.0087C21.2578 20.2572 24.6437 23.7013 28.8203 23.7013Z" fill="#EED655"/>
<path d="M48.0302 29.5214C49.7235 29.5214 51.0961 28.1256 51.0961 26.4038C51.0961 24.6819 49.7235 23.2861 48.0302 23.2861C46.337 23.2861 44.9644 24.6819 44.9644 26.4038C44.9644 28.1256 46.337 29.5214 48.0302 29.5214Z" fill="#EED655"/>
<path d="M39.0412 33.2628C41.4118 33.2628 43.3335 31.3087 43.3335 28.8981C43.3335 26.4876 41.4118 24.5334 39.0412 24.5334C36.6707 24.5334 34.749 26.4876 34.749 28.8981C34.749 31.3087 36.6707 33.2628 39.0412 33.2628Z" fill="#EED655"/>
<path d="M14.309 32.8493C17.6955 32.8493 20.4407 30.0568 20.4407 26.6121C20.4407 23.1674 17.6955 20.3749 14.309 20.3749C10.9225 20.3749 8.17725 23.1674 8.17725 26.6121C8.17725 30.0568 10.9225 32.8493 14.309 32.8493Z" fill="#EED655"/>
<path d="M26.7764 42.8293C29.8242 42.8293 32.295 40.3161 32.295 37.2158C32.295 34.1156 29.8242 31.6023 26.7764 31.6023C23.7286 31.6023 21.2578 34.1156 21.2578 37.2158C21.2578 40.3161 23.7286 42.8293 26.7764 42.8293Z" fill="#EED655"/>
<path d="M5.92739 43.2424C8.07216 43.2424 9.81084 41.4743 9.81084 39.2934C9.81084 37.1124 8.07216 35.3444 5.92739 35.3444C3.78262 35.3444 2.04395 37.1124 2.04395 39.2934C2.04395 41.4743 3.78262 43.2424 5.92739 43.2424Z" fill="#EED655"/>
<path d="M15.9441 59.0461C20.2337 59.0461 23.711 55.5089 23.711 51.1456C23.711 46.7823 20.2337 43.2451 15.9441 43.2451C11.6546 43.2451 8.17725 46.7823 8.17725 51.1456C8.17725 55.5089 11.6546 59.0461 15.9441 59.0461Z" fill="#EED655"/>
<path d="M11.4483 81.0838C16.4151 81.0838 20.4415 76.9882 20.4415 71.9359C20.4415 66.8836 16.4151 62.788 11.4483 62.788C6.48148 62.788 2.45508 66.8836 2.45508 71.9359C2.45508 76.9882 6.48148 81.0838 11.4483 81.0838Z" fill="#EED655"/>
<path d="M2.86148 51.1432C4.44184 51.1432 5.72297 49.8404 5.72297 48.2334C5.72297 46.6264 4.44184 45.3236 2.86148 45.3236C1.28113 45.3236 0 46.6264 0 48.2334C0 49.8404 1.28113 51.1432 2.86148 51.1432Z" fill="#EED655"/>
<path d="M15.7397 99.7953C19.9164 99.7953 23.3022 96.3512 23.3022 92.1027C23.3022 87.8542 19.9164 84.4102 15.7397 84.4102C11.5631 84.4102 8.17725 87.8542 8.17725 92.1027C8.17725 96.3512 11.5631 99.7953 15.7397 99.7953Z" fill="#EED655"/>
<path d="M26.5712 112.685C29.9577 112.685 32.7029 109.893 32.7029 106.448C32.7029 103.003 29.9577 100.211 26.5712 100.211C23.1847 100.211 20.4395 103.003 20.4395 106.448C20.4395 109.893 23.1847 112.685 26.5712 112.685Z" fill="#EED655"/>
<path d="M26.7783 75.6769C28.4715 75.6769 29.8441 74.2811 29.8441 72.5593C29.8441 70.8375 28.4715 69.4417 26.7783 69.4417C25.085 69.4417 23.7124 70.8375 23.7124 72.5593C23.7124 74.2811 25.085 75.6769 26.7783 75.6769Z" fill="#EED655"/>
<path d="M29.8407 85.6556C31.8726 85.6556 33.5197 83.9806 33.5197 81.9144C33.5197 79.8482 31.8726 78.1732 29.8407 78.1732C27.8088 78.1732 26.1616 79.8482 26.1616 81.9144C26.1616 83.9806 27.8088 85.6556 29.8407 85.6556Z" fill="#EED655"/>
<path d="M37.2001 99.7962C40.5866 99.7962 43.3318 97.0037 43.3318 93.559C43.3318 90.1143 40.5866 87.3218 37.2001 87.3218C33.8136 87.3218 31.0684 90.1143 31.0684 93.559C31.0684 97.0037 33.8136 99.7962 37.2001 99.7962Z" fill="#EED655"/>
<path d="M51.7125 112.686C56.1149 112.686 59.6838 109.056 59.6838 104.577C59.6838 100.099 56.1149 96.469 51.7125 96.469C47.3101 96.469 43.7412 100.099 43.7412 104.577C43.7412 109.056 47.3101 112.686 51.7125 112.686Z" fill="#EED655"/>
<path d="M38.8358 118.921C40.8677 118.921 42.5148 117.246 42.5148 115.179C42.5148 113.113 40.8677 111.438 38.8358 111.438C36.8039 111.438 35.1567 113.113 35.1567 115.179C35.1567 117.246 36.8039 118.921 38.8358 118.921Z" fill="#EED655"/>
<path d="M47.8258 121C49.4062 121 50.6873 119.697 50.6873 118.09C50.6873 116.483 49.4062 115.18 47.8258 115.18C46.2455 115.18 44.9644 116.483 44.9644 118.09C44.9644 119.697 46.2455 121 47.8258 121Z" fill="#EED655"/>
<path d="M69.0863 24.533C73.6016 24.533 77.262 20.8096 77.262 16.2167C77.262 11.6237 73.6016 7.90039 69.0863 7.90039C64.571 7.90039 60.9106 11.6237 60.9106 16.2167C60.9106 20.8096 64.571 24.533 69.0863 24.533Z" fill="#EED655"/>
<path d="M72.3562 6.65095C74.1623 6.65095 75.6265 5.16209 75.6265 3.32548C75.6265 1.48887 74.1623 0 72.3562 0C70.5501 0 69.0859 1.48887 69.0859 3.32548C69.0859 5.16209 70.5501 6.65095 72.3562 6.65095Z" fill="#EED655"/>
<path d="M81.9612 10.3932C84.3317 10.3932 86.2534 8.43907 86.2534 6.02851C86.2534 3.61796 84.3317 1.66382 81.9612 1.66382C79.5906 1.66382 77.6689 3.61796 77.6689 6.02851C77.6689 8.43907 79.5906 10.3932 81.9612 10.3932Z" fill="#EED655"/>
<path d="M93.6131 19.959C96.7738 19.959 99.3361 17.3526 99.3361 14.1376C99.3361 10.9225 96.7738 8.31616 93.6131 8.31616C90.4524 8.31616 87.8901 10.9225 87.8901 14.1376C87.8901 17.3526 90.4524 19.959 93.6131 19.959Z" fill="#EED655"/>
<path d="M104.443 37.0085C108.846 37.0085 112.415 33.3783 112.415 28.9001C112.415 24.422 108.846 20.7917 104.443 20.7917C100.041 20.7917 96.4722 24.422 96.4722 28.9001C96.4722 33.3783 100.041 37.0085 104.443 37.0085Z" fill="#EED655"/>
<path d="M108.944 58.6301C114.249 58.6301 118.55 54.2552 118.55 48.8585C118.55 43.4617 114.249 39.0868 108.944 39.0868C103.638 39.0868 99.3374 43.4617 99.3374 48.8585C99.3374 54.2552 103.638 58.6301 108.944 58.6301Z" fill="#EED655"/>
<path d="M82.9857 32.8493C86.3722 32.8493 89.1175 30.0568 89.1175 26.6121C89.1175 23.1674 86.3722 20.3749 82.9857 20.3749C79.5993 20.3749 76.854 23.1674 76.854 26.6121C76.854 30.0568 79.5993 32.8493 82.9857 32.8493Z" fill="#EED655"/>
<path d="M90.7513 42.8267C92.7832 42.8267 94.4304 41.1517 94.4304 39.0855C94.4304 37.0193 92.7832 35.3444 90.7513 35.3444C88.7194 35.3444 87.0723 37.0193 87.0723 39.0855C87.0723 41.1517 88.7194 42.8267 90.7513 42.8267Z" fill="#EED655"/>
<path d="M93.6096 51.5592C95.4157 51.5592 96.8799 50.0703 96.8799 48.2337C96.8799 46.3971 95.4157 44.9082 93.6096 44.9082C91.8035 44.9082 90.3394 46.3971 90.3394 48.2337C90.3394 50.0703 91.8035 51.5592 93.6096 51.5592Z" fill="#EED655"/>
<path d="M104.24 77.7581C108.755 77.7581 112.416 74.0347 112.416 69.4418C112.416 64.8488 108.755 61.1255 104.24 61.1255C99.7248 61.1255 96.0645 64.8488 96.0645 69.4418C96.0645 74.0347 99.7248 77.7581 104.24 77.7581Z" fill="#EED655"/>
<path d="M72.1504 118.924C77.4558 118.924 81.7568 114.549 81.7568 109.152C81.7568 103.755 77.4558 99.3804 72.1504 99.3804C66.8449 99.3804 62.5439 103.755 62.5439 109.152C62.5439 114.549 66.8449 118.924 72.1504 118.924Z" fill="#EED655"/>
<path d="M91.9774 113.517C96.4927 113.517 100.153 109.794 100.153 105.201C100.153 100.608 96.4927 96.8849 91.9774 96.8849C87.4621 96.8849 83.8018 100.608 83.8018 105.201C83.8018 109.794 87.4621 113.517 91.9774 113.517Z" fill="#EED655"/>
<path d="M106.692 100.212C110.079 100.212 112.824 97.4194 112.824 93.9747C112.824 90.53 110.079 87.7375 106.692 87.7375C103.306 87.7375 100.561 90.53 100.561 93.9747C100.561 97.4194 103.306 100.212 106.692 100.212Z" fill="#EED655"/>
<path d="M114.462 86.4869C116.72 86.4869 118.55 84.6258 118.55 82.3301C118.55 80.0343 116.72 78.1732 114.462 78.1732C112.204 78.1732 110.374 80.0343 110.374 82.3301C110.374 84.6258 112.204 86.4869 114.462 86.4869Z" fill="#EED655"/>
<path d="M117.526 76.0911C119.445 76.0911 121.001 74.5091 121.001 72.5577C121.001 70.6063 119.445 69.0244 117.526 69.0244C115.607 69.0244 114.052 70.6063 114.052 72.5577C114.052 74.5091 115.607 76.0911 117.526 76.0911Z" fill="#EED655"/>
<path d="M93.6117 91.0638C96.9982 91.0638 99.7435 88.2713 99.7435 84.8266C99.7435 81.3819 96.9982 78.5894 93.6117 78.5894C90.2253 78.5894 87.48 81.3819 87.48 84.8266C87.48 88.2713 90.2253 91.0638 93.6117 91.0638Z" fill="#EED655"/>
<path d="M82.1679 96.0513C84.4256 96.0513 86.2557 94.1902 86.2557 91.8944C86.2557 89.5986 84.4256 87.7375 82.1679 87.7375C79.9103 87.7375 78.0801 89.5986 78.0801 91.8944C78.0801 94.1902 79.9103 96.0513 82.1679 96.0513Z" fill="#EED655"/>
<path d="M72.9687 98.1303C74.6619 98.1303 76.0346 96.7345 76.0346 95.0127C76.0346 93.2908 74.6619 91.895 72.9687 91.895C71.2755 91.895 69.9028 93.2908 69.9028 95.0127C69.9028 96.7345 71.2755 98.1303 72.9687 98.1303Z" fill="#EED655"/>
</svg>

    </>
  )
}

export default Logo