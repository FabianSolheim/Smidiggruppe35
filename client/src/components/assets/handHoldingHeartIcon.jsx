import { SvgIcon } from "@mui/material";

export function HandHoldingHeartIcon(props) {
  return (
    <SvgIcon {...props}>
      <svg
        width="20"
        height="20"
        viewBox="0 0 71 71"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M35.0728 35.039C35.5832 35.6605 36.4144 35.6605 36.9321 35.039L44.8723 25.4484C47.1763 22.6602 47.0451 18.0413 44.464 15.4715C42.2183 13.2292 38.8716 13.6323 36.8155 16.1181L35.9988 17.1007L35.1895 16.1265C33.1334 13.6323 29.7867 13.2292 27.541 15.4715C24.9672 18.0413 24.8286 22.6602 27.1327 25.4484L35.0728 35.039ZM56.2175 41.5559C55.3571 40.6573 54.0156 40.7161 53.1114 41.5559L46.3743 47.7621C45.5504 48.5263 44.5223 48.9378 43.4578 48.9378H34.8322C34.1906 48.9378 33.6656 48.3332 33.6656 47.5941C33.6656 46.8551 34.1906 46.2505 34.8322 46.2505H40.5413C41.7006 46.2505 42.7797 45.3351 42.9693 44.0166C43.2099 42.3369 42.087 40.8757 40.6652 40.8757H28.9992C27.0306 40.8757 25.1276 41.6567 23.5964 43.0844L20.206 46.2505H16.1666C15.525 46.2505 15 46.8551 15 47.5941V55.6563C15 56.3953 15.525 57 16.1666 57H42.1818C43.2391 57 44.2671 56.5885 45.0983 55.8243L56.1227 45.6626C57.231 44.6464 57.3185 42.6981 56.2175 41.5559Z"
          fill="#034F7A"
        />
        <path
          d="M71 35.5C71 55.1061 55.1061 71 35.5 71C15.8939 71 0 55.1061 0 35.5C0 15.8939 15.8939 0 35.5 0C55.1061 0 71 15.8939 71 35.5ZM7.1 35.5C7.1 51.1849 19.8151 63.9 35.5 63.9C51.1849 63.9 63.9 51.1849 63.9 35.5C63.9 19.8151 51.1849 7.1 35.5 7.1C19.8151 7.1 7.1 19.8151 7.1 35.5Z"
          fill="#034F7A"
        />
      </svg>
    </SvgIcon>
  );
}
