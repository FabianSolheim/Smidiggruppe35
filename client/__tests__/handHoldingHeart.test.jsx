import { HandHoldingHeartIcon } from "../src/components/assets/handHoldingHeartIcon";
import ReactDOM from "react-dom";
import React from "react";

describe("HandHoldingHeartIcon test", () => {
  it("shows HandHoldingHeartIcon", () => {
    const domElement = document.createElement("svg");
    ReactDOM.render(<HandHoldingHeartIcon />, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
