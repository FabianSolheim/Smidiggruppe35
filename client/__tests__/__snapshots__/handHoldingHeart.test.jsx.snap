// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandHoldingHeartIcon test shows HandHoldingHeartIcon 1`] = `"<svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\"><svg width=\\"20\\" height=\\"20\\" viewBox=\\"0 0 71 71\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M35.0728 35.039C35.5832 35.6605 36.4144 35.6605 36.9321 35.039L44.8723 25.4484C47.1763 22.6602 47.0451 18.0413 44.464 15.4715C42.2183 13.2292 38.8716 13.6323 36.8155 16.1181L35.9988 17.1007L35.1895 16.1265C33.1334 13.6323 29.7867 13.2292 27.541 15.4715C24.9672 18.0413 24.8286 22.6602 27.1327 25.4484L35.0728 35.039ZM56.2175 41.5559C55.3571 40.6573 54.0156 40.7161 53.1114 41.5559L46.3743 47.7621C45.5504 48.5263 44.5223 48.9378 43.4578 48.9378H34.8322C34.1906 48.9378 33.6656 48.3332 33.6656 47.5941C33.6656 46.8551 34.1906 46.2505 34.8322 46.2505H40.5413C41.7006 46.2505 42.7797 45.3351 42.9693 44.0166C43.2099 42.3369 42.087 40.8757 40.6652 40.8757H28.9992C27.0306 40.8757 25.1276 41.6567 23.5964 43.0844L20.206 46.2505H16.1666C15.525 46.2505 15 46.8551 15 47.5941V55.6563C15 56.3953 15.525 57 16.1666 57H42.1818C43.2391 57 44.2671 56.5885 45.0983 55.8243L56.1227 45.6626C57.231 44.6464 57.3185 42.6981 56.2175 41.5559Z\\" fill=\\"#490374\\"></path><path d=\\"M71 35.5C71 55.1061 55.1061 71 35.5 71C15.8939 71 0 55.1061 0 35.5C0 15.8939 15.8939 0 35.5 0C55.1061 0 71 15.8939 71 35.5ZM7.1 35.5C7.1 51.1849 19.8151 63.9 35.5 63.9C51.1849 63.9 63.9 51.1849 63.9 35.5C63.9 19.8151 51.1849 7.1 35.5 7.1C19.8151 7.1 7.1 19.8151 7.1 35.5Z\\" fill=\\"#490374\\"></path></svg></svg>"`;
